src/string_ptr.rs

```diff
@@ -39,21 +44,38 @@ impl Write<String> for StringPtr {
                 .unwrap(),
         )?;
         write_str(offset, value, env)?;
+
+        // pin
+        let pin = export_asr!(fn_pin, env);
+        pin.call(&[Value::I32(offset.try_into().unwrap())])
+            .expect("Failed to call __pin");
+
         Ok(Box::new(StringPtr::new(offset)))
     }
 
-    fn write(&self, value: &str, env: &Env) -> anyhow::Result<()> {
+    fn write(&mut self, value: &str, env: &Env) -> anyhow::Result<Box<StringPtr>> {
```

Comment:
Not quite sure about making self mutable , but I did not see any way around

Reply:
It completely made sense that a `write` needs a mutable reference on self!

Reply:
Indeed x)